lib/suse/connect/client.rb

```diff
@@ -25,10 +26,43 @@ def initialize(config)
       def register!
         announce_or_update
         product = @config.product || Zypper.base_product
+
+        register_product(product, install_release_package: @config.product ? true : false)
```

Comment:
Seems like you are trying to call `register_product` as if it was a method with keyword arguments.
However, it isn't -- it is defined with regular arguments:
```
def register_product(product, install_release_package = true)
```
...as opposed to:
```
def register_product(product, install_release_package: true)
```

So when you call it like this, the second parameter `install_release_package` will be a hash: `{:install_release_package=>false}` -- and it will always be truthy.

Reply:
good catch. fixing this